rollup/internal/controller/watcher/chunk_proposer.go

```diff
@@ -145,6 +154,45 @@ func (p *ChunkProposer) updateDBChunkInfo(chunk *encoding.Chunk, codecVersion en
 		return nil
 	}
 
+	compatibilityBreachOccurred := false
+
+	for {
+		compatible, err := encoding.CheckChunkCompressedDataCompatibility(chunk, codecVersion)
+		if err != nil {
+			log.Error("Failed to check chunk compressed data compatibility", "start block number", chunk.Blocks[0].Header.Number, "codecVersion", codecVersion, "err", err)
+			return err
+		}
+
+		if compatible {
+			break
+		}
+
+		compatibilityBreachOccurred = true
+
+		if len(chunk.Blocks) == 1 {
+			log.Warn("Disable compression: cannot truncate chunk with only 1 block for compatibility", "block number", chunk.Blocks[0].Header.Number)
+			break
+		}
+
+		chunk.Blocks = chunk.Blocks[:len(chunk.Blocks)-1]
```

Comment:
shouldn't we here also recompute the msg queue hash?

Reply:
True. So does `batch.Chunks = batch.Chunks[:len(batch.Chunks)-1]`, maybe these violations are never triggered.

Reply:
Fixed in [edbe6f6](https://github.com/scroll-tech/scroll/pull/1658/commits/edbe6f6659f1075f9a116a7152c9211bdf73e685).